index.js

```diff
@@ -175,6 +189,10 @@ Filter.prototype.build = function() {
       return result;
     }));
   }).then(function(result) {
+    return Promise.all(pendingWork).then(function() {
```

Comment:
If any async work rejects, we will unexpectedly complete early and orphan the pending work. We will need to wait until pending work is complete before continuing regardless of anything fails.

We could also explore some cancellation, but that may require some more brain storming.